src/swap/switchain.js

```diff
@@ -0,0 +1,371 @@
+        fromCurrencyCode = 'BTC'
+        apiUrl = 'https://api-testnet.switchain.com/rest/v1'
+      } else {
+        apiUrl = 'https://api.switchain.com/rest/v1'
+      }
+
+      const pair = `${fromCurrencyCode.toUpperCase()}-${toCurrencyCode.toUpperCase()}`
+
+      // get wallet addresses for exchange
+      const [fromAddress, toAddress] = await Promise.all([
+        getAddress(fromWallet, fromCurrencyCode),
+        getAddress(toWallet, toCurrencyCode)
+      ])
+
+      // Check for supported currencies, even if we aren't activated:
+      const json: SwitchainOfferResponse = await swHttpCall(
+        '/offer',
+        'GET',
+        null,
+        {
+          pair,
+          orderIdSeed: toAddress
+        }
+      )
+      const {
+        maxLimit,
+        minLimit,
+        minerFee,
+        quote,
+        signature,
+        expiryTs,
+        orderId
+      } = json
+
+      // get native min max limits
+      const [nativeMax, nativeMin] = await Promise.all([
+        fromWallet.denominationToNative(maxLimit.toString(), fromCurrencyCode),
+        fromWallet.denominationToNative(minLimit.toString(), fromCurrencyCode)
+      ])
+
+      // determine amount param for order creation
+      const [fromAmount, toAmount] = await Promise.all([
+        fromWallet.nativeToDenomination(nativeAmount, fromCurrencyCode),
+        toWallet.nativeToDenomination(nativeAmount, toCurrencyCode)
+      ])
+
+      quoteForFrom = quoteFor === 'from'
+      // check for min / max limits
+      if (quoteForFrom) {
+        if (lt(nativeAmount, nativeMin)) {
+          throw new SwapBelowLimitError(swapInfo, nativeMin)
+        }
+        if (gt(nativeAmount, nativeMax)) {
+          throw new SwapAboveLimitError(swapInfo, nativeMax)
+        }
+      } else {
+        const toAmountInFrom = div(toAmount, quote, 8)
+
+        if (lt(toAmountInFrom, minLimit)) {
+          throw new SwapBelowLimitError(swapInfo, nativeMin)
+        }
+
+        if (gt(toAmountInFrom, maxLimit)) {
+          throw new SwapAboveLimitError(swapInfo, nativeMax)
+        }
+      }
+
+      // order creation body
+      const quoteAmount = quoteForFrom ? { fromAmount } : { toAmount }
+      orderCreationBody = {
+        pair,
+        toAddress,
+        refundAddress: fromAddress,
+        signature,
+        orderId,
+        ...quoteAmount
+      }
+
+      // plugin output creation
+      fromNativeAmount = nativeAmount
+      toNativeAmount = await toWallet.denominationToNative(
+        sub(mul(quote, fromAmount), minerFee),
+        toCurrencyCode
+      )
+      if (!quoteForFrom) {
+        fromNativeAmount = await fromWallet.denominationToNative(
+          div(add(toAmount, minerFee), quote, 8),
+          fromCurrencyCode
+        )
+        toNativeAmount = nativeAmount
+      }
+
+      const destinationAddress = toAddress
+      const expirationDate = new Date(expiryTs * 1000)
+      const isEstimate = false
```

Comment:
This is always a constant, so there is no need to pass it into your `makeSwapPluginQuote`.

Reply:
Not entirely sure what you meant here :)

Reply:
One of the parameters to `makeSwapPluginQuote` is `isEstimate`. However, `isEstimate` is always false. Therefore, you could just delete that parameter, and put the `false` directly inside `makeSwapPluginQuote`.